ext/crypto/00_crypto.js

```diff
@@ -469,19 +478,30 @@
       });
 
       // 2.
-      if (ArrayBufferIsView(keyData)) {
-        keyData = new Uint8Array(
-          keyData.buffer,
-          keyData.byteOffset,
-          keyData.byteLength,
-        );
+      if (format !== "jwk") {
+        if (ArrayBufferIsView(keyData) || (keyData instanceof ArrayBuffer)) {
+          if (ArrayBufferIsView(keyData)) {
+            keyData = new Uint8Array(
+              keyData.buffer,
+              keyData.byteOffset,
+              keyData.byteLength,
+            );
+          } else {
+            keyData = new Uint8Array(keyData);
+          }
+          keyData = TypedArrayPrototypeSlice(keyData);
+        } else {
+          throw new TypeError("keyData is a JsonWebKey");
+        }
       } else {
-        keyData = new Uint8Array(keyData);
+        if (ArrayBufferIsView(keyData) || (keyData instanceof ArrayBuffer)) {
+          throw new TypeError("keyData is not a JsonWebKey");
+        }
       }
-      keyData = TypedArrayPrototypeSlice(keyData);
 
       const normalizedAlgorithm = normalizeAlgorithm(algorithm, "importKey");
 
+      // 2.
```

Comment:
This should be moved into the `case "HMAC"` below (it is defined in the "HMAC Import Key Operation" in the spec).

Reply:
done